### PR TITLE
Add supcon seg for regression

### PR DESCRIPTION
**Pull request by sungmanc**
### Summary

Issue number: https://github.com/openvinotoolkit/training_extensions/issues/2171

This PR includes the supcon tests for regression. 

### How to test

tox -e tests-all-py310 -- tests/regression/semantic_segmentation/test_segmentation.py::TestRegressionSupconSegmentation
image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+"""Initialization of OCR-Lite-HRnet-18-mod2 model for SupCon Segmentation Task."""
```
Comment (sungchul1) [inline otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/__init__.py]: OCR-Lite-HRnet-18-mod2 -> OCR-Lite-HRnet-18

File: otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/model.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+"""Model configuration of OCR-Lite-HRnet-18-mod2 model for SupCon Segmentation Task."""
```
Comment (sungchul1) [inline otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/model.py]: OCR-Lite-HRnet-18-mod2 -> OCR-Lite-HRnet-18

**Review by sungchul1 — COMMENTED** [review]
LGTM. I left some minor comments.

File: otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/__init__.py (hunk 1 above)
Comment (sungmanc) [inline otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/__init__.py]:
```suggestion
"""Initialization of OCR-Lite-HRnet-18 model for SupCon Segmentation Task."""
```

File: otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/model.py (hunk 2 above)
Comment (sungmanc) [inline otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/model.py]:
```suggestion
"""Model configuration of OCR-Lite-HRnet-18 model for SupCon Segmentation Task."""

```

File: otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/__init__.py (hunk 1 above)
Comment (sungmanc) [reply otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/__init__.py]: Done.

File: otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/model.py (hunk 2 above)
Comment (sungmanc) [reply otx/algorithms/segmentation/configs/ocr_lite_hrnet_18/supcon/model.py]: Done.

**Review by sungchul1 — DISMISSED** [review]

**Review by jaegukhyun — DISMISSED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Review by sungchul1 — APPROVED** [review]

**Pull request merged**